apps/blockchain/contracts/starknet/src/erc721_bridgable.cairo

```diff
@@ -0,0 +1,260 @@
+        _name::write(name_);
+        _symbol::write(symbol_);
+    }
+
+    #[internal]
+    fn _owner_of(token_id: u256) -> Option<ContractAddress> {
+        let owner = _owners::read(token_id);
+        match owner.is_zero() {
+            bool::False(()) => Option::Some(owner),
+            bool::True(()) => Option::None(())
+        }
+    }
+
+    #[internal]
+    fn _exists(token_id: u256) -> bool {
+        !_owners::read(token_id).is_zero()
+    }
+
+    #[internal]
+    fn _is_approved_or_owner(spender: ContractAddress, token_id: u256) -> bool {
+        let owner = _owner_of(token_id).expect('ERC721: invalid token ID');
+        owner == spender | is_approved_for_all(owner, spender) | spender == get_approved(token_id)
+    }
+
+    #[internal]
+    fn _approve(to: ContractAddress, token_id: u256) {
+        let owner = _owner_of(token_id).expect('ERC721: invalid token ID');
+        assert(owner != to, 'ERC721: approval to owner');
+        _token_approvals::write(token_id, to);
+        Approval(owner, to, token_id);
+    }
+
+    #[internal]
+    fn _set_approval_for_all(owner: ContractAddress, operator: ContractAddress, approved: bool) {
+        assert(owner != operator, 'ERC721: self approval');
+        _operator_approvals::write((owner, operator), approved);
+        ApprovalForAll(owner, operator, approved);
+    }
+
+    #[internal]
+    fn _mint(to: ContractAddress, token_id: u256) {
+        assert(!to.is_zero(), 'ERC721: invalid receiver');
+        assert(!_exists(token_id), 'ERC721: token already minted');
+
+        // Update balances
+        _balances::write(to, _balances::read(to) + 1.into());
+
+        // Update token_id owner
+        _owners::write(token_id, to);
+
+        // Emit event
+        Transfer(Zeroable::zero(), to, token_id);
+    }
+
+    #[internal]
+    fn _transfer(from: ContractAddress, to: ContractAddress, token_id: u256) {
+        assert(!to.is_zero(), 'ERC721: invalid receiver');
+        let owner = _owner_of(token_id).expect('ERC721: invalid token ID');
+        assert(from == owner, 'ERC721: wrong sender');
+
+        // Implicit clear approvals, no need to emit an event
+        _token_approvals::write(token_id, Zeroable::zero());
+
+        // Update balances
+        _balances::write(from, _balances::read(from) - 1.into());
+        _balances::write(to, _balances::read(to) + 1.into());
+
+        // Update token_id owner
+        _owners::write(token_id, to);
+
+        // Emit event
+        Transfer(from, to, token_id);
+    }
+
+    #[internal]
+    fn _burn(token_id: u256) {
+        let owner = _owner_of(token_id).expect('ERC721: invalid token ID');
+
+        // Implicit clear approvals, no need to emit an event
+        _token_approvals::write(token_id, Zeroable::zero());
+
+        // Update balances
+        _balances::write(owner, _balances::read(owner) - 1.into());
+
+        // Delete owner
+        _owners::write(token_id, Zeroable::zero());
```

Comment:
Would it be interesting to keep track of the burnt token? Because burnt mean that it can never be minted again, right?

Because if we only write 0 we can't do the difference between a burnt token or an un-mint token.

Perhaps it's fine as in both cases there is no owner.

I don't have experience on that. Just wanted to have a confirmation.

Reply:
No need to attach too much importance to this contract, as we will be taking the OpenZeppelin version when it is released in Cairo 1.0 (for this one, I took inspiration from the WIP version): https://github.com/OpenZeppelin/cairo-contracts/tree/erc721/src/openzeppelin/token

However, indeed, we can bring up your remark to them as it seems justified. Perhaps Martin from OpenZeppelin could explain this implementation choice for the burn